src/compiler/parser.ts

```diff
@@ -2143,16 +2166,31 @@ module ts {
             }
         }
 
+        function setContextParameters(explicitFlags: ParserContextFlags = ParserContextFlags.None, ambientFlags: ParserContextFlags = ParserContextFlags.None) {

```

Comment:
Yes, that would be good. I think it would be even better to remove this function entirely.